tests/test_pytypes.cpp

```diff
@@ -923,4 +926,41 @@ TEST_SUBMODULE(pytypes, m) {
 #else
     m.attr("defined_PYBIND11_TYPING_H_HAS_STRING_LITERAL") = false;
 #endif
+
+#if defined(PYBIND11_HAS_RANGES) // test_ranges
+    m.attr("defined_PYBIND11_HAS_RANGES") = true;
```

Comment:
I'd move this to the end if this `#if` block, so that it is right over the `false` assignment. For readability.